config.m4

```diff
@@ -217,7 +217,6 @@ if test "$MONGODB" != "no"; then
     bson-timegm.c \
     bson-utf8.c \
     bson-value.c \
-    bson-version.c \
```

Comment:
Good catch. mongoc-version-functions.c is also a new file in the rc0 and needs to be included